pkg/virt-launcher/virtwrap/manager.go

```diff
@@ -293,15 +293,7 @@ func (l *LibvirtDomainManager) KillVMI(vmi *v1.VirtualMachineInstance) error {
 		log.Log.Object(vmi).Info("Domain stopped.")
 	}
 
-	err = dom.Undefine()
-	if err != nil {
-		if domainerrors.IsNotFound(err) {
-			return nil
-		}
-		log.Log.Object(vmi).Reason(err).Error("Undefining the domain state failed.")
-		return err
-	}
-	log.Log.Object(vmi).Info("Domain undefined.")
+	log.Log.Object(vmi).Info("Domain not running or paused, nothing to do.")
```

Comment:
the domain still needs to be undefined. The domain informer needs this. virt-launcher also waits on the undefine before exiting I believe.

Reply:
At least the pods are disappearing as expected... but I will check if I find something which waits for domain deletion.

Reply:
I think that virt-launcher only waits for the qemu process to disappear.

Reply:
virt-launcher waits for the deletion notification to be sent to virt-handler before exiting.

This is to ensure that under normal operation, virt-handler correctly receives the reason that the qemu pid exiting. Virt-launcher is essentially acking that virt-handler has been notified about the deletion before allowing the pod to go away.

If the deletion notification does not occur, the Pod will still terminate eventually, but virt-handler will not know that the Domain disappeared until the watchdog file for that VM expires.